src/main/java/com/clouway/adapter/InputStreamAdapterConcrete.java

```
@@ -0,0 +1,25 @@
+package com.clouway.adapter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Created by Petar Nedelchev <peter.krasimirov@gmail.com>.
+ */
+public class InputStreamAdapterConcrete implements InputStreamAdapter {
+    private InputStream input;
+
+    public InputStreamAdapterConcrete(InputStream input){
+        this.input = input;
+    }
+
+    public int read(byte[] buf, int offset, int length) {
+        int readBytes = -1;
+        try {
+            readBytes = input.read(buf, offset, length);
+        } catch (IOException e) {
```

do not catch this exception this way you hide the error from the user of the adapter